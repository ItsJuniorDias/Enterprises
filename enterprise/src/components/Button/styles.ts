import styled from 'styled-components/native';
import { RectButton } from 'react-native-gesture-handler';

export const Container = styled(RectButton)`
  width: 100%;
  height: 60px;
  background-color: #376aed;
  border-radius: 10px;
  justify-content: center;
  align-items: center;
  margin-top: 54px;
`;

export const ButtonText = styled.Text`
  font-family: 'Poppins-Medium';
  color: #fff;
  font-size: 18px;
`;
