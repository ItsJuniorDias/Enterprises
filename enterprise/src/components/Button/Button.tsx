import React from 'react';
import { RectButtonProperties } from 'react-native-gesture-handler';

import { Container, ButtonText } from './styles';

interface ButtonProps extends RectButtonProperties {
  children: string;
}

const Button = ({ children, ...rest }: ButtonProps) => {
  return (
    <Container {...rest}>
      <ButtonText>{children}</ButtonText>
    </Container>
  );
};

export default Button;
