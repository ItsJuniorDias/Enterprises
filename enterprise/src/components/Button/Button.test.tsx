import React, { Component } from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import Button from '../Button/Button';

jest.mock('react-native-gesture-handler', () => ({
  RectButton: jest.fn().mockReturnValue(() => jest.fn()),
}));

describe('Behavior Button', () => {
  it('should render button', () => {
    const result = render(<Button children="Entrar" />).toJSON();

    expect(result).toMatchSnapshot();
  });
});
