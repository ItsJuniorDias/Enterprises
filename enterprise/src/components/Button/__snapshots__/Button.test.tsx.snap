// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior Button should render button 1`] = `null`;
